game-app/game-core/src/main/java/games/strategy/triplea/odds/calculator/BattleCalculatorPanel.java

```diff
@@ -1111,6 +1095,16 @@
           }
         }
 
+        // Mimic the legacy behavior and force the parties to be non-allied modifying either
+        // attacker or defender like the old code did). This legacy behavior was requested in #9532.
+        if (data.getRelationshipTracker().isAllied(getDefender(), getAttacker())) {
```

Comment:
Is this if statement a tautology? When the battle calculator loads initially, I observed the attacker and defender were the same. Hence, checking if an attacker is allied with defender (same player) would always be true.

Reply:
As I explained in https://github.com/triplea-game/triplea/pull/9532#issuecomment-908161318, the behaviour you describe happens if the current player selects its own territory.  It does not occur when the battle calculator is initiated on a territory belonging to a foreign player, e.g. current player is Carthage and (without moving pieces) you select a territory of the Greek City states, then Carthage is the attacker and the City States are the defender, in particular they are not allied.  Moreover, we do not want to override the City States with Rom, which would be the first entry in the list of enemies of Carthage.  Hence, this guard.